File: com/white/RemoveDuplicatesFromSortedArray.java
package com.white;

public class RemoveDuplicatesFromSortedArray {

    // 26. Remove Duplicates from Sorted Array
    public static int removeDuplicates(int[] nums) {
        if (nums==null || nums.length==0) {
            return 0;
        }
        if (nums.length==1) {
            return 1;
        }

        int currNum = nums[0];
        int currLastIndex = 0;
        int currIndex = 0;
        while(currIndex < nums.length) {
            if (nums[currIndex]==currNum) {
                currIndex++;
            } else {
                currLastIndex++;
                nums[currLastIndex] = nums[currIndex];
                currNum = nums[currIndex];
                currIndex++;
            }
        }

        return currLastIndex+1;
    }
}

File: LeetcodeGo/RemoveDuplicatesFromSortedArray.go
package main

// 26. Remove Duplicates from Sorted Array
func removeDuplicates(nums []int) int {
    uniqueCount := 1
    currNum := nums[0]

    for i := 1; i < len(nums); i++ {
        if nums[i] == currNum {
            nums[i] = 200
        } else {
            uniqueCount++
            currNum = nums[i]
        }
    }

    placeIndex := 0
    for j := 0; j < len(nums); j++ {
        if nums[j] != 200 {
            nums[placeIndex] = nums[j]
            placeIndex++
        }
    }

    return uniqueCount
}